### DISCUSSION TITLE
add expiresIn in s3 getSignedUrl doesn't work.

### DISCUSSION THREAD

**Discussion by tsulatsitamim** [adonisjs/core#3430] [Help]
how to set expires in s3 signed url? I try the docs but I get only 15 minutes, which is the default value.

```
await Drive.use('s3').getSignedUrl(path, {
    expiresIn: '30min',
  })
```

**Comment by alex-semenyuk** [comment]
See example here https://docs.adonisjs.com/guides/auth/api-tokens-guard

So try `30mins` instead of `30min`

**Comment by thetutlage** [comment]
`30mins` and `30min` should make no difference.

It depends upon which AWS credentials you have used to generate the signed URL. If it is `IAM instance profile`, then this token itself has a validity of 6 hours and if you generate the URL lets say after **5 hours and 45 mins**, then the token will make the URL expiry before its original expiry.

AWS has many moving parts and usually inconsistencies are created because of them. You can learn more about it here https://stackoverflow.com/a/57792699
